### PR TITLE
python3Packages.craft-archives: 2.0.0 -> 2.0.2

### PR DESCRIPTION
**Pull request by jnsgruk**
Bump to latest upstream tag, 2.0.2. Also bumps craft-parts to 2.1.4.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jnsgruk** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363994`

---
### `x86_64-linux`
<details>
  <summary>:x: 14 packages failed to build:</summary>
  <ul>
    <li>charmcraft</li>
    <li>charmcraft.dist</li>
    <li>python311Packages.craft-application</li>
    <li>python311Packages.craft-application.dist</li>
    <li>python311Packages.craft-archives</li>
    <li>python311Packages.craft-archives.dist</li>
    <li>python312Packages.craft-application</li>
    <li>python312Packages.craft-application.dist</li>
    <li>python312Packages.craft-archives</li>
    <li>python312Packages.craft-archives.dist</li>
    <li>rockcraft</li>
    <li>rockcraft.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
  </ul>
</details>

**Comment by jnsgruk** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363994`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>charmcraft</li>
    <li>charmcraft.dist</li>
    <li>python311Packages.craft-application</li>
    <li>python311Packages.craft-application.dist</li>
    <li>python311Packages.craft-archives</li>
    <li>python311Packages.craft-archives.dist</li>
    <li>python311Packages.craft-parts</li>
    <li>python311Packages.craft-parts.dist</li>
    <li>python312Packages.craft-application</li>
    <li>python312Packages.craft-application.dist</li>
    <li>python312Packages.craft-archives</li>
    <li>python312Packages.craft-archives.dist</li>
    <li>python312Packages.craft-parts</li>
    <li>python312Packages.craft-parts.dist</li>
    <li>rockcraft</li>
    <li>rockcraft.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/craft-archives/default.nix [hunk 1]
```diff
@@ -43,6 +44,7 @@ buildPythonPackage rec {
     lazr-restfulclient
     overrides
     pydantic
+    debian
```
Comment (dotlambda) [inline pkgs/development/python-modules/craft-archives/default.nix:47]:
```suggestion
    python-debian
```
Please rename this package to what it's called on PyPI.

File: pkgs/development/python-modules/craft-archives/default.nix [hunk 2]
```diff
@@ -43,6 +44,7 @@ buildPythonPackage rec {
...
     tabulate
```
Comment (dotlambda) [inline pkgs/development/python-modules/craft-archives/default.nix:48]:
```suggestion
```
but two other dependencies are missing from the list

File: pkgs/development/python-modules/craft-parts/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/craft-parts/default.nix:1]: urllib3 should be removed

File: pkgs/development/python-modules/craft-archives/default.nix (hunk 1 above)
Comment (jnsgruk) [reply pkgs/development/python-modules/craft-archives/default.nix:47]: Just so I understand, you mean add another commit that actually renames the package in nixpkgs? Or do something differently here?

Looks like the pname is already set to `python-debian`, it's just in a directory named `debian`, not `python-debian`.

Comment (dotlambda) [reply pkgs/development/python-modules/craft-archives/default.nix]: > Just so I understand, you mean add another commit that actually renames the package in nixpkgs?

Exactly.



> Looks like the pname is already set to `python-debian`, it's just in a directory named `debian`, not `python-debian`.

More importantly, the attribute in `python-packages.nix` needs to be renamed.

**Comment by dotlambda** [comment]
Please add an entry to `python-aliases.nix`.

**Comment by jnsgruk** [comment]
> Please add an entry to `python-aliases.nix`.

Done in https://github.com/NixOS/nixpkgs/pull/363994/commits/3fb3a0cbe80f012990e11322a14d0ea6f7da3cc9

**Comment by jnsgruk** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363994`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>charmcraft</li>
    <li>charmcraft.dist</li>
    <li>python311Packages.craft-application</li>
    <li>python311Packages.craft-application.dist</li>
    <li>python311Packages.craft-archives</li>
    <li>python311Packages.craft-archives.dist</li>
    <li>python311Packages.craft-parts</li>
    <li>python311Packages.craft-parts.dist</li>
    <li>python311Packages.python-debian</li>
    <li>python311Packages.python-debian.dist</li>
    <li>python312Packages.craft-application</li>
    <li>python312Packages.craft-application.dist</li>
    <li>python312Packages.craft-archives</li>
    <li>python312Packages.craft-archives.dist</li>
    <li>python312Packages.craft-parts</li>
    <li>python312Packages.craft-parts.dist</li>
    <li>python312Packages.python-debian</li>
    <li>python312Packages.python-debian.dist</li>
    <li>rockcraft</li>
    <li>rockcraft.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
  </ul>
</details>

**Review by dotlambda — APPROVED** [review]

**Comment by jnsgruk** [comment]
Thanks for the help @dotlambda 🥳

**Pull request merged**